File: app/src/main/java/com/absoluteapps/arthurl/mobooru/Sub.java
package com.absoluteapps.arthurl.mobooru;

import java.io.Serializable;

/**
 * Created by pspka_000 on 9/10/2015.
 */
public class Sub implements Serializable, Comparable {
    public String subName;
    public int subID = 0;
    public int subscriberCount = 0;
    public boolean selected = false;
    public boolean isNSFW = false;
    public boolean isCustom = false;
    public String desc;

    public Sub(String subName, int subID, int subscriberCount, boolean selected, boolean isNSFW, boolean isCustom, String desc) {
        this.subName = subName;
        this.subID = subID;
        this.subscriberCount = subscriberCount;
        this.selected = selected;
        this.isNSFW = isNSFW;
        this.isCustom = isCustom;
        this.desc = desc;
    }

    public Sub(String subName, int id) {
        this.subName = subName.toLowerCase();
        this.subID = id;
    }

    public Sub(String subName, int id, int subscriberCount) {
        this.subName = subName.toLowerCase();
        this.subID = id;
        this.subscriberCount = subscriberCount;
    }

    public Sub() {

    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public boolean isNSFW() {
        return isNSFW;
    }

    public void setNSFW(boolean NSFW) {
        isNSFW = NSFW;
    }

    public int compareTo(Object s2) {
        return ((Sub) this).subName.compareTo(((Sub) s2).subName);
    }
}

File: app/src/main/java/com/absoluteapps/arthurl/mobooru/Sub.kt
package com.absoluteapps.arthurl.mobooru

import java.io.Serializable

/**
 * Created by pspka_000 on 9/10/2015.
 */
open class Sub : Serializable, Comparable<Sub> {
    override fun compareTo(other: Sub): Int {
        return when {
            this.subName > other.subName -> 1
            this.subName < other.subName -> -1
            else -> 0
        }
    }

    var subName = ""
    var subscriberCount = 0
    var selected = false
    var isNSFW = false
    var isCustom = false
    var desc = ""

    constructor(subName: String, subscriberCount: Int, selected: Boolean, isNSFW: Boolean, isCustom: Boolean, desc: String) {
        this.subName = subName
        this.subscriberCount = subscriberCount
        this.selected = selected
        this.isNSFW = isNSFW
        this.isCustom = isCustom
        this.desc = desc
    }

    constructor(subName: String) {
        this.subName = subName.toLowerCase()
    }

    constructor(subName: String, subscriberCount: Int) {
        this.subName = subName.toLowerCase()
        this.subscriberCount = subscriberCount
    }

    constructor()
}
